### PR TITLE
[Backport] Trim email address in newsletter, forgot password, checkout login and email to a friend form

### PR DESCRIPTION
**Pull request by gelanivishal**
### Original Pull Request 
 https://github.com/magento/magento2/pull/16564

Related to issue: #6058 

### Description

Trim email address by remove leading or trailing space on following forms.
- Newsletter
- Checkout login
- Forgot your password
- Email to a Friend

### Fixed Issues (if relevant)

1. magento/magento2#6058: IE11 user login email validation fails if field has leading or trailing space

### Manual testing scenarios

1. Open forgot your password page in Firefox or IE browser.
2. Try to add space before entering an email address in the Email field.
3. Copy `" johndoe@domain.com "` and paste in the Email field. It will automatically remove leading or trailing space.
4. Check same on the footer newsletter form
5. Check same by adding a product to cart and go to checkout page without login. Check an Email Address field under the Shipping Address section.
6. After login, go to the product view page and go to Email to a Friend form. Check an Email field under Sender and Invitee sections.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by VladimirZaets** [comment]
Hi @gelanivishal, thank you for collaboration.
We tested your PR and we found the next problem:

**Problem**: Subscribe field is missed on Home Page.

Problem is caused by changes in `app/code/Magento/Newsletter/view/frontend/templates/subscribe.phtml`

**Comment by gelanivishal** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @gelanivishal. Thank you for your request.
Unfortunately, I can only deploy instances for `2.2-develop` and `2.3-develop` base branches

**Comment by gelanivishal** [comment]
@VladimirZaets Please check now.

**Comment by VladimirZaets** [comment]
Hi @gelanivishal, we found the next problem after manual testing:
**Problems**: Email field is missed on customer/account/forgotpassword page
16713forgotpassword

**Comment by gelanivishal** [comment]
@VladimirZaets Please check now.

**Comment by gelanivishal** [comment]
@VladimirZaets Please check now.

File: app/code/Magento/SendFriend/view/frontend/templates/send.phtml [hunk 1]
```diff
@@ -73,6 +74,9 @@
             <div class="control">
                 <input name="sender[email]" value="<?php echo $block->escapeHtml($block->getEmail()) ?>"
                        title="<?php /* @escapeNotVerified */ echo __('Email') ?>" id="sender-email" type="text" class="input-text"
+                <input name="sender[email]" value="<?= $block->escapeHtmlAttr($block->getEmail()) ?>"
```
Comment (VladimirZaets) [inline app/code/Magento/SendFriend/view/frontend/templates/send.phtml:77]: The method `escapeHtmlAttr` is absent in the `\Magento\SendFriend\Block\Send` in the Magento 2.1 version

**Comment by sidolov** [comment]
Hi @gelanivishal , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**